addons/social_share/controllers/main.py

```diff
@@ -0,0 +1,104 @@
+import base64
+
+from odoo.http import Controller, content_disposition, request, route
+
+from .user_agents import NETWORK_TO_AGENT
+from ..utils.image_utils import scale_image
+
+class SocialShareController(Controller):
+
+    @route(['/snshare/cards/<int:campaign_id>/card.png', '/snshare/cards/<int:campaign_id>/<string:uuid>/card.png'], type='http', auth='public', sitemap=False, website=True)
+    def snshare_campaign_image(self, campaign_id, uuid=None, small=False):
+        campaign_sudo = request.env['snshare.campaign'].sudo().browse(campaign_id).exists()
+        url_sudo = request.env['snshare.url'].sudo()
+        target = None
+        if campaign_sudo.model_id:
+            if not uuid:
+                return request.not_found()
```

Comment:
In the end, if model becomes required, we will be able to force uuid and simplify the route, isn't it ?